rosbag2_py/rosbag2_py/_storage.pyi

```diff
@@ -53,14 +54,24 @@ class QoS:
     def avoid_ros_namespace_conventions(self, arg0: bool) -> QoS: ...
     def best_effort(self) -> QoS: ...
     def deadline(self, arg0: Duration) -> QoS: ...
+    def depth(self) -> int: ...
+    @overload
+    def durability(self) -> rmw_qos_durability_policy_t: ...
+    @overload
```

Comment:
So the more we add to this, the more I wonder why we have this structure at all; it is essentially duplicating what is in https://github.com/ros2/rclpy/blob/rolling/rclpy/rclpy/qos.py .

@MichaelOrlov Can you explain more about this structure, and why we are using this rather than `rclpy.QoSProfile`?

Reply:
My guess here, it's that QoS on rclpy is a Python class not a pybind11 wrapper class. Make impossible to return this type, that's why it's duplicated. is this assumption right ?

Reply:
@clalancette Yes. basically, because this is a pybind11 wrapper class https://github.com/ros2/rosbag2/pull/1648/files#diff-6fb0986aa5f23a37be72468378f8959ba9702304ec4504d318a8c7602211c625R191-R206 sort of converter to the `rclcpp::qos` class.  
I don't know for sure, but I think it is because the all other functions in the rosbag2_py are pybind11 wrappers and need to convert returning or incoming values to/from the `rclcpp::qos` class.

Reply:
@MichaelOrlov Just stumbled upon the issue when trying to adapt for the new API changes in our ros2bag_tools.

Two questions that I have is it not possible to just use a typecaster or otherwise directly map rclcpp QoS or make rclpy QoSProfile somehow a direct bind?

As a side note I still think storage QoS is not publically imported.

Reply:
@devrite I am sorry, I don't have a better solution on top of my head. However, constructive suggestions and contributions are welcome.

As regards
> As a side note I still think storage QoS is not publically imported.

Could you please elaborate on that?

Reply:
Have not worked with oybibd enough to be sure it works but either way, if QoSProfiles can not be replaced by the QoS bind of the current PR on the long run a converter function on the python side or pybind side (typecaster) may be useful.

Regarding my last comment. In the import list above I do not see the QoS type listed. Meaning if somebody wants to use it, when creating a TopicInfo object, you would need to import it currently from _storage and not the public module.